web/src/test/java/com/replica/RouterServerTest.java

```diff
@@ -0,0 +1,346 @@
+        yaml.registerModule(new GraphHopperConfigModule());
+        JsonNode yamlNode = yaml.readTree(new File(TEST_GRAPHHOPPER_CONFIG_PATH));
+        graphHopperConfiguration = yaml.convertValue(yamlNode.get("graphhopper"), GraphHopperConfig.class);
+        ObjectMapper json = Jackson.newObjectMapper();
+        GraphHopperManaged graphHopperManaged = new GraphHopperManaged(graphHopperConfiguration, json);
+        graphHopperManaged.start();
+        return graphHopperManaged;
+    }
+
+    private static void startTestServer() throws Exception {
+        // Load Graphhopper using already-built graph files
+        GraphHopperManaged graphHopperManaged = loadGraphhopper();
+
+        // Grab instances of auto/bike/ped router and PT router (if applicable)
+        GraphHopper graphHopper = graphHopperManaged.getGraphHopper();
+        PtRouter ptRouter = null;
+        if (graphHopper instanceof GraphHopperGtfs) {
+            ptRouter = new PtRouterImpl(graphHopper.getTranslationMap(), graphHopper.getGraphHopperStorage(),
+                    graphHopper.getLocationIndex(), ((GraphHopperGtfs) graphHopper).getGtfsStorage(),
+                    RealtimeFeed.empty(((GraphHopperGtfs) graphHopper).getGtfsStorage()),
+                    graphHopper.getPathDetailsBuilderFactory());
+        }
+
+        // Create matrix API instance
+        MatrixAPI matrixAPI = new GHMatrixAPI(graphHopper, graphHopperConfiguration);
+
+        // Load GTFS link mapping and GTFS info maps for use in building responses
+        Map<String, String> gtfsLinkMappings = null;
+        Map<String, List<String>> gtfsRouteInfo = null;
+        Map<String, String> gtfsFeedIdMapping = null;
+
+        File linkMappingsDbFile = new File("transit_data/gtfs_link_mappings.db");
+        if (linkMappingsDbFile.exists()) {
+            DB db = DBMaker.newFileDB(linkMappingsDbFile).readOnly().make();
+            gtfsLinkMappings = db.getHashMap("gtfsLinkMappings");
+            gtfsRouteInfo = db.getHashMap("gtfsRouteInfo");
+            gtfsFeedIdMapping = db.getHashMap("gtfsFeedIdMap");
+        }
+
+        // Start in-process test server + instantiate stub
+        String uniqueName = InProcessServerBuilder.generateName();
+        InProcessServerBuilder.forName(uniqueName)
+                .directExecutor() // directExecutor is fine for unit tests
+                .addService(new RouterImpl(graphHopper, ptRouter, matrixAPI, gtfsLinkMappings,
+                        gtfsRouteInfo, gtfsFeedIdMapping, null, TEST_REGION_NAME))
+                .addService(ProtoReflectionService.newInstance())
+                .build().start();
+        ManagedChannel channel = InProcessChannelBuilder.forName(uniqueName)
+                .directExecutor()
+                .build();
+
+        routerStub = router.RouterGrpc.newBlockingStub(channel);
+    }
+
+    @BeforeAll
+    public static void setUp() throws Exception {
```

Comment:
@mark-idleman - It might be worth factoring this out into some test utils or something, so it can be reused in other tests?

Reply:
Yeah, I think that's definitely a good idea, I was just debating doing it now vs. in the next change because we still don't have very many tests. But, might be nice to just combine now, I'll see how much I can move to a shared test util(s)

Reply:
Refactored and pulled out all shared logic into a new class that I had the individual tests extend from. This also meant that I moved the existing tests @danielhfrank added recently for the street export to using mini_kc's OSM + GTFS, instead of the original testing files; I'm not removing those files, though, because they could be useful in the future if we want to test weird GTFS edge cases or something